kodi_addon_checker/addons/Repository.py

```diff
@@ -16,26 +18,59 @@
 from ..versions import AddonVersion
 
 
+class RateLimitedAdapter(requests.adapters.HTTPAdapter):
+    def __init__(self, *args, retries=5, wait=10, **kwargs):
+        self._last_send = None
+        self._wait_time = wait
+        max_retries = requests.adapters.Retry(
+            total=retries,
+            backoff_factor=wait,
+            status_forcelist={429, },
+            allowed_methods=None,
+        )
+        kwargs.setdefault('max_retries', max_retries)
+        super().__init__(*args, **kwargs)
+
+    def send(self, *args, **kwargs):
+        if self._last_send:
+            delta = time.time() - self._last_send
+            if delta < self._wait_time:
+                time.sleep(self._wait_time - delta)
+
+        self._last_send = time.time()
+        response = super().send(*args, **kwargs)
+        status_code = getattr(response, 'status_code', None)
+        if 300 <= status_code < 400:
+            self._last_send = None
+        return response
+
+
 class Repository():
+    # Recover from unreliable mirrors
+    _session = requests.Session()
+    _adapter = RateLimitedAdapter(retries=5, wait=10)
+    _session.mount('http://', _adapter)
+    _session.mount('https://', _adapter)
+    atexit.register(_session.close)
+
     def __init__(self, version, path):
         super().__init__()
         self.version = version
         self.path = path
+        self.addons = []
```

Comment:
Maybe this line should be moved back to where it was, so that the checker will fail if the versioned repository download fails due to some other server error that cannot be resolved by the allowed retries?